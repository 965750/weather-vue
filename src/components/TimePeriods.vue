<template>
    <v-row>
        <v-col
            :class="{'text-center': true, 'pa-1': true, 'period__block--divider': index !== 0}"
            v-for="(time, index) in periods" :key="time.time"
        >
            <p class="my-0 caption  font-weight-bold">
                {{ Math.round(time.main.temp) | kelvinsToCelsius }} &#8451;
            </p>
            <p class="period__label my-0 caption">
                {{
                    checkHour(time.time, index) === 'now' ?
                    $t(checkHour(time.time, index)) :
                    checkHour(time.time, index)
                }}
            </p>
        </v-col>
    </v-row>
</template>

<script>
export default {
  name: 'TimePeriods',
  props: {
    periods: {
      required: true,
      type: Array,
    },
  },
  methods: {
    checkHour(time, index) {
      const date = new Date(time * 1000);

      if (index === 0) {
        return 'now';
      }

      return `${date.getHours() + 1}:00`;
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../assets/main';

.period {
    &__block {
        &--divider {
            border-left: 1px solid $aluminium;
        }
    }

    &__label {
        color: $aluminium;
        text-transform: uppercase;
    }
}
</style>
