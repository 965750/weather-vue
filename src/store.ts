import Vue from 'vue';
import Vuex from 'vuex';
import axios from 'axios';
import config from '@/config';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    lang: 'en',
    card: null,
  },
  mutations: {
    setCard(state, data) {
      const weatherList = data.list.map((weather: any) => ({
        clouds: weather.clouds.all,
        time: weather.dt,
        wind: weather.wind.speed,
        main: {
          humidity: weather.main.humidity,
          pressure: weather.main.pressure,
          temp: weather.main.temp,
        },
      }));

      state.card = {
        city: {
          name: data.city.name,
          country: data.city.country,
        },
        weatherList,
      };
    },
  },
  actions: {
    citySearch({ commit }, city: string) {
      axios
        .get(`https://api.openweathermap.org/data/2.5/forecast?q=${city}&APPID=${config.apiKey}`)
        .then((res) => {
          commit('setCard', res.data);
        });
    },
  },
});
