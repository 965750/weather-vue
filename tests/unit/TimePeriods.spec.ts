import {
  shallowMount,
  createLocalVue,
} from '@vue/test-utils';
import VueI18n from 'vue-i18n';
import TimePeriods from '@/components/TimePeriods.vue';
import filters from '@/myFilters';
import store from '@/store';
import messages from '@/locale/languages';

const localVue = createLocalVue();
localVue.use(VueI18n);

const i18n = new VueI18n({
  locale: store.getters.getLanguage,
  messages,
});

filters.forEach((filter) => {
  localVue.filter(filter.name, filter.handler);
});

const initialProps = {
  periods: [
    {
      main: {
        temp: 22,
      },
      time: 168999000,
    },
    {
      main: {
        temp: 24,
      },
      time: 168991000,
    },
  ],
};

describe('TimePeriods.vue', () => {
  const createWrapper = (props = initialProps) => shallowMount(TimePeriods, {
    i18n,
    localVue,
    propsData: {
      ...props,
    },
    stubs: {
      'v-col': true,
      'v-row': true,
    },
  });

  it('should match initial snapshot with valid hour format', () => {
    const wrapper = createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });
});
