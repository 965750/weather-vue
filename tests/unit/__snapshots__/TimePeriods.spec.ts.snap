// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePeriods.vue should match initial snapshot with valid hour format 1`] = `
<v-row-stub>
  <v-col-stub
    class="text-center pa-1"
  >
    <p
      class="my-0 caption  font-weight-bold"
    >
      
            -251 ℃
        
    </p>
     
    <p
      class="period__label text-uppercase my-0 caption"
    >
      
            teraz
        
    </p>
  </v-col-stub>
  <v-col-stub
    class="text-center pa-1 period__block--divider"
  >
    <p
      class="my-0 caption  font-weight-bold"
    >
      
            -249 ℃
        
    </p>
     
    <p
      class="period__label text-uppercase my-0 caption"
    >
      
            23:00
        
    </p>
  </v-col-stub>
</v-row-stub>
`;
